src/tool_operate.c

```diff
@@ -697,45 +697,59 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
   return result;
 }
 
+/* helper function to ensure input ends in val_to_ensure */
+static void ensure_trailing(char **input, const char val_to_ensure)
+{
+  if(*input && **input) {
+    size_t len = strlen(*input);
+    if(((*input)[len - 1] != val_to_ensure)) {
+      char *composed_input = aprintf("%s%c", *input, val_to_ensure);
+      if(composed_input) {
```

Comment:
When this `aprintf()` call fails, shouldn't `*input` be freed and pointed to `""` or NULL or something?

Reply:
That does mean every usage of this function needs to be checked `if(!ensure_trailing(..)) // error handling`.
I'll turn it into a function returning a CURLcode and checking for that on each invocation of this function.